restorers/model/mprnet/resize.py

```diff
@@ -0,0 +1,129 @@
+from typing import Optional
+
+import tensorflow as tf
+
+
+class DownSample(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        channels: int,
+        channel_factor: float,
+        name: str = "DownSample Block",
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(name=name, *args, **kwargs)
+
+        self.channels = channels
+        self.channel_factor = channel_factor
+
+        self.downsample = tf.keras.Sequential(
+            [
+                tf.keras.layers.AveragePooling2D(pool_size=2, strides=2),
```

Comment:
I have some doubts as to whether `AveragePooling2D` is an accuracte approximation of [this op](https://github.com/swz30/MPRNet/blob/main/Denoising/MPRNet.py#L158) from the official implementation or not. This could also be written as something like
```python
y = tf.image.resize(
    x, size=[tf.shape(x)[1] // 2, tf.shape(x)[2] // 2], method="bilinear"
)
```
However, I'm not sure if this would work and not encounter `None`s as spatial dimentions.
@SauravMaheshkar We need to experiment and figure this out.

Reply:
The current error is most probably caused by the downsampling implementation. Tried this didn't work. Will need more thought.

Reply:
used a bunch of `assert` statements to realize that the error is actually in the `ChannelAttentionBlock`. Rather than being `isotonic` (not changing the shape) somehow it is misbehaving within the `Encoder` but not in a standalone manner.

Reply:
```
>       assert x.shape == inputs.shape, f"{x.shape} != {inputs.shape}"
E       AssertionError: Exception encountered when calling layer 'encoder_layer_2_0' (type ChannelAttentionBlock).
E       
E       (1, 64, 64, 228) != (1, 64, 64, 8640)
E       
E       Call arguments received by layer 'encoder_layer_2_0' (type ChannelAttentionBlock):
E         • inputs=tf.Tensor(shape=(1, 64, 64, 8640), dtype=float32)
E         • training=False
```

The `ChannelAttentionBlock` is eating 🍴  up dimensions somehow. The confounding thing thing is it isn't misbehaving during the test meant for the `ChannelAttentionBlock` but only when it's being used within the `Encoder`